akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -83,10 +86,10 @@ private[http2] class Http2ServerDemux(http2Settings: Http2ServerSettings, initia
     new GraphStageLogic(shape) with Http2MultiplexerSupport with Http2StreamHandling with GenericOutletSupport with StageLogging {
       logic =>
 
-      override def settings: Http2ServerSettings = http2Settings
+      override def settings: Http2CommonSettings = http2Settings
       override def isUpgraded: Boolean = upgraded
 
-      override protected def logSource: Class[_] = classOf[Http2ServerDemux]
+      override protected def logSource: Class[_] = if (isServer) classOf[Http2ServerDemux] else classOf[Http2ClientDemux]
```

Comment:
`isServer` only used here so far to be able to distinguish logs when running server and client at the same time in tests.